python/src/opendp/mod.py

```diff
@@ -1377,6 +1361,37 @@ def _exception_predicate(v):
     return signed_band_search(center, at_center, sign)
 
 
+def _infer_type(predicate: Callable[[float], bool]) -> Union[Type[float], Type[int]]:
+    def _is_type_error(e):
+        is_match_error = isinstance(e, OpenDPException) and "No match for concrete type" in (e.message or "")
```

Comment:
This worries me: Someone making a tweak to the error message where it is thrown isn't going to expect that the exact value matters.

Reply:
If necessary, a new error class, so you can catch it without looking at the message?

Reply:
Yes, plumbing error classes would be a good thing. This is working and non-critical though.